Bug: The compilation of mod-congrats-on-level encountered an error and failed to compile.

### Current Behaviour

I have updated to the latest version of azerothcore-wotlk and mod-eluna. After adding mod-congrats-on-level, I encountered errors during the compilation process.

The hash value of azerothcore-wotlk is: 0c099a75ec709f30308b4b2a07c9befd450e9134

`[ 26%] Building CXX object modules/CMakeFiles/modules.dir/mod-eluna/src/LuaEngine/ElunaInstanceAI.cpp.o
/root/azerothcore/modules/mod-congrats-on-level/src/mod_congratsonlevel.cpp:138:57: fatal error: only virtual member functions can be marked 'override'
    void OnLevelChanged(Player* player, uint8 oldLevel) override
                                                        ^~~~~~~~
1 error generated.
make[2]: *** [modules/CMakeFiles/modules.dir/build.make:258: modules/CMakeFiles/modules.dir/mod-congrats-on-level/src/mod_congratsonlevel.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 26%] Building CXX object src/server/apps/CMakeFiles/authserver.dir/authserver/Authentication/AuthCodes.cpp.o
`


### Expected Behaviour

I hope to expedite the resolution of this compilation error.

### Steps to reproduce the problem

1. Install azerothcore-wotlk #21238
2. Install mod-eluna #261
3. Install mod-congrats-on-level #35
An error occurs during compilation, and the build cannot be completed.

### Extra Notes

If any additional explanation is needed, please let me know.

### AC rev. hash/commit

commit 0c099a75ec709f30308b4b2a07c9befd450e9134 (HEAD -> master, origin/master, origin/HEAD)
Author: Paul <60552737+demetrzz@users.noreply.github.com>
Date:   Mon Feb 17 01:27:45 2025 +0300

    fix(Core/Spell): Corrected values for event food (#21465)

    Co-authored-by: pavel_k <pavel_k@mail.com>

### Operating system

Ubuntu 22.04.5 LTS (GNU/Linux 5.15.0-131-generic x86_64)

### Custom changes or Modules

This is a fresh installation, and no modifications have been made.

--- a/src/mod_congratsonlevel.cpp
+++ b/src/mod_congratsonlevel.cpp
@@ -64,9 +64,10 @@ config file for quick modifications.
 */
 
 #include "Configuration/Config.h"
-#include "ScriptMgr.h"
-#include "Player.h"
 #include "Chat.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+#include "WorldSessionMgr.h"
 
 struct COL
 {
@@ -249,7 +250,7 @@ public:
                     default:
                         break;
                 }
-                sWorld->SendServerMessage(SERVER_MSG_STRING, ss.str().c_str());
+                sWorldSessionMgr->SendServerMessage(SERVER_MSG_STRING, ss.str().c_str());
             }
 
             // If level is defined, they hit a reward level.
@@ -278,7 +279,7 @@ public:
                     default:
                         break;
                 }
-                sWorld->SendServerMessage(SERVER_MSG_STRING, ss.str().c_str());
+                sWorldSessionMgr->SendServerMessage(SERVER_MSG_STRING, ss.str().c_str());
 
                 // Issue a raid warning to the player
                 std::ostringstream ss2;

--- a/src/mod_congratsonlevel.cpp
+++ b/src/mod_congratsonlevel.cpp
@@ -117,25 +117,27 @@ class CongratsAnnounce : public PlayerScript
 
 public:
 
-    CongratsAnnounce() : PlayerScript("CongratsAnnounce") {}
+    CongratsAnnounce() : PlayerScript("CongratsAnnounce", {
+        PLAYERHOOK_ON_LOGIN
+    }) {}
 
-    void OnLogin(Player* player)
+    void OnPlayerLogin(Player* player)
     {
         // Announce Module
         if (col.congratsAnnounce)
-        {
             ChatHandler(player->GetSession()).SendSysMessage(col.acoreMessageId);
-        }
     }
 };
 
 class CongratsOnLevel : public PlayerScript
 {
 public:
-    CongratsOnLevel() : PlayerScript("CongratsOnLevel") { }
+    CongratsOnLevel() : PlayerScript("CongratsOnLevel", {
+        PLAYERHOOK_ON_LEVEL_CHANGED
+    }) { }
 
     // Level Up Rewards
-    void OnLevelChanged(Player* player, uint8 oldLevel) override
+    void OnPlayerLevelChanged(Player* player, uint8 oldLevel) override
     {
         // If enabled...
         if (col.congratsEnable)
@@ -148,72 +150,56 @@ public:
                 case 10:
                 {
                     if (oldLevel < 10)
-                    {
                         money = giveAward(player);
-                    }
                 }
                 break;
 
                 case 20:
                 {
                     if (oldLevel < 20)
-                    {
                         money = giveAward(player);
-                    }
                 }
                 break;
 
                 case 30:
                 {
                     if (oldLevel < 30)
-                    {
                         money = giveAward(player);
-                    }
                 }
                 break;
 
                 case 40:
                 {
                     if (oldLevel < 40)
-                    {
                         money = giveAward(player);
-                    }
                 }
                 break;
 
                 case 50:
                 {
                     if (oldLevel < 50)
-                    {
                         money = giveAward(player);
-                    }
                 }
                 break;
 
                 case 60:
                 {
                     if (oldLevel < 60)
-                    {
                         money = giveAward(player);
-                    }
                 }
                 break;
 
                 case 70:
                 {
                     if (oldLevel < 70)
-                    {
                         money = giveAward(player);
-                    }
                 }
                 break;
 
                 case 80:
                 {
                     if (oldLevel < 80)
-                    {
                         money = giveAward(player);
-                    }
                 }
                 break;
 
@@ -311,7 +297,9 @@ public:
 class ModCongratsLevelWorldScript : public WorldScript
 {
 public:
-    ModCongratsLevelWorldScript() : WorldScript("ModCongratsLevelWorldScript") { }
+    ModCongratsLevelWorldScript() : WorldScript("ModCongratsLevelWorldScript", {
+        WORLDHOOK_ON_BEFORE_CONFIG_LOAD
+    }) { }
 
     void OnBeforeConfigLoad(bool reload) override
     {

--- a/src/mod_congratsonlevel.cpp
+++ b/src/mod_congratsonlevel.cpp
@@ -150,56 +150,72 @@ public:
                 case 10:
                 {
                     if (oldLevel < 10)
+                    {
                         money = giveAward(player);
+                    }
                 }
                 break;
 
                 case 20:
                 {
                     if (oldLevel < 20)
+                    {
                         money = giveAward(player);
+                    }
                 }
                 break;
 
                 case 30:
                 {
                     if (oldLevel < 30)
+                    {
                         money = giveAward(player);
+                    }
                 }
                 break;
 
                 case 40:
                 {
                     if (oldLevel < 40)
+                    {
                         money = giveAward(player);
+                    }
                 }
                 break;
 
                 case 50:
                 {
                     if (oldLevel < 50)
+                    {
                         money = giveAward(player);
+                    }
                 }
                 break;
 
                 case 60:
                 {
                     if (oldLevel < 60)
+                    {
                         money = giveAward(player);
+                    }
                 }
                 break;
 
                 case 70:
                 {
                     if (oldLevel < 70)
+                    {
                         money = giveAward(player);
+                    }
                 }
                 break;
 
                 case 80:
                 {
                     if (oldLevel < 80)
+                    {
                         money = giveAward(player);
+                    }
                 }
                 break;
 
@@ -237,7 +253,7 @@ public:
             }
 
             // If level is defined, they hit a reward level.
-            if (!level && col.CongratsPerLevelEnable)
+            if (!level)
             {
                 // Issue a server notification for the player on level up.
                 std::ostringstream ss;

--- a/src/mod_congratsonlevel.cpp
+++ b/src/mod_congratsonlevel.cpp
@@ -253,7 +253,7 @@ public:
             }
 
             // If level is defined, they hit a reward level.
-            if (!level)
+            if (!level && col.CongratsPerLevelEnable)
             {
                 // Issue a server notification for the player on level up.
                 std::ostringstream ss;

